dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/repositories/DataRequestRepository.kt

```diff
@@ -82,4 +88,32 @@ interface DataRequestRepository : JpaRepository<DataRequestEntity, String> {
     fun searchDataRequestEntity(
         @Param("searchFilter") searchFilter: GetDataRequestsSearchFilter,
     ): List<DataRequestEntity>
+
+    /** This method updates the Request Status to Answered for an open request with a specific framework,
+     * reporting period as well as company identifier as soon as the (manual) QA accepted the provided data
+     * by the corresponding company.
+     * @param dataRequestCompanyIdentifierValue to check for
+     * @param reportingPeriod to check for
+     * @param dataTypeName to check for
+     * @returns the aggregated data requests
+     */
+    @Transactional
+    @Modifying
+    @Query
+    (
+        "UPDATE DataRequestEntity d " +
+            "SET d.requestStatus = " +
+            ":#{T(org.dataland.datalandcommunitymanager.model.dataRequest.RequestStatus).Answered} " +
+            "WHERE " +
+            "(d.dataTypeName = :#{#dataTypeName} AND " +
+            "d.requestStatus = :#{T(org.dataland.datalandcommunitymanager.model.dataRequest.RequestStatus).Open} " +
+            "AND " +
+            "d.reportingPeriod = :#{#reportingPeriod} AND " +
+            "d.dataRequestCompanyIdentifierValue = :#{#dataRequestCompanyIdentifierValue})",
+    )
+    fun updateDataRequestEntitiesFromOpenToAnswered(
+        dataRequestCompanyIdentifierValue: String,
+        reportingPeriod: String,
+        dataTypeName: String,
+    )
```

Comment:
Does this function update ALL stored DataRequestEntity objects for which the three properties (identifier value, reporting period, framework) match?

Reply:
Isn't it a bit redundant to have a function here in the repository that updates the status from open to answered and to also keep the `patchDataRequest` function in the SingleDataRequestManager, which is also designed to update the status of a data request? The difference between the two is that this one here gets identifier value, reporting period and framework as inputs while the `patchDataRequest` eats a dataRequestId and a certain status, meaning that it is able to not only set the status from open -> answered but to make the transition between all available stati; however, only for the one dataRequest corresponding to the provided Id. Do you think their requirements are different enough to have two functions changing the status?

Reply:
I would say so. One is precisely targeting one request and change its field as a user wishes. The other one searches all requests and marks the found ones as answered. Doing two or more subsequent queries in the latter case feels inefficient to me. And joining the two queries would be quite messy. If you have a suggestion, I'm happy tho hear 😄